### PR TITLE
basic: add note that "has" does not do mathematics

### PR DESCRIPTION
**Pull request by cbm755**
Related to Issue #10573: nothing to fix but hopefully this helps
someone else in the future.

### CODE REVIEW

**Comment by cbm755** [comment]
@asmeurer this is related to [your comment](https://github.com/sympy/sympy/issues/10573#issuecomment-183385561) some time ago.

Do let me know if this is too verbose so such a minor thing.

File: sympy/core/basic.py [hunk 1]
```diff
@@ -1132,6 +1132,30 @@ def has(self, *patterns):
         >>> x.has(x)
         True
 
+        Note ``has`` is a structural algorithm with no knowledge of
+        mathematics. Assuming otherwise can lead to what seem to be
```
Comment (smichr) [inline sympy/core/basic.py]: I would keep it brief with one good example. Maybe just the Interval example, showing the interval so one can see that it is open at the end points:

```
...mathematics. Consider the following interval:

>>> from sympy.sets import Interval
>>> i = Interval.Lopen(0, 5); i
(0, 5]
>>> i.has(4)  # there is no "4" in the arguments
False
>>> i.has(0)  # there *is* a "0" in the arguments
True

The use of `contains` will tell whether a number is in the interval or not, however:

>>> i.contains(4)
True
>>> i.contains(0)
False
```

**Comment by cbm755** [comment]
Thanks @smichr that's much better, I used it with minor rewording.

**Comment by smichr** [comment]
Thanks @cbm755. I think examples like this are good to have to help resolve "gotchas" at the source.

**Pull request merged**